homeassistant/components/wemo.py

```diff
@@ -79,23 +80,53 @@ def discovery_dispatch(service, discovery_info):
 
     discovery.listen(hass, SERVICE_WEMO, discovery_dispatch)
 
-    _LOGGER.info("Scanning for WeMo devices.")
-    devices = [(device.host, device) for device in pywemo.discover_devices()]
+    def setup_url_for_device(device):
+        """Determine setup.xml url for given device."""
+        return 'http://{}:{}/setup.xml'.format(device.host, device.port)
+
+    def setup_url_for_address(address):
+        """Determine setup.xml url for given address."""
+        host, _, port = address.partition(':')
 
-    # Add static devices from the config file.
-    devices.extend((address, None)
-                   for address in config.get(DOMAIN, {}).get(CONF_STATIC, []))
+        if port:
+            port = cv.port(port)
```

Comment:
Break out the port extraction from address and port validation to a validator function that we use in the config schema. We want to fail as early as possible if config is bad, during config validation, not after.